jwst/assign_wcs/util.py

```diff
@@ -275,18 +275,39 @@ def is_fits(input):
 
 def subarray_transform(input_model):
     """
-    Inputs are in full frame coordinates.
-    If a subarray observation - shift the inputs.
+    Return an offset model if the observation uses a subarray.
 
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.DataModel`
+        Data model.
+
+    Returns
+    -------
+    subarray2full : `~astropy.modeling.core.Model` or ``None``
+        Returns a (combination of ) ``Shift`` models is a subarray is used.
+        Returns ``None`` if a full frame observation.
     """
+    tr_xstart = astmodels.Identity(1)
+    tr_ystart = astmodels.Identity(1)
+
+    # These quanities are 1-based
     xstart = input_model.meta.subarray.xstart
     ystart = input_model.meta.subarray.ystart
-    if xstart is None:
-        xstart = 1
-    if ystart is None:
-        ystart = 1
-    subarray2full = astmodels.Shift(xstart - 1) & astmodels.Shift(ystart - 1)
-    return subarray2full
+
```

Comment:
This section is because in the data, xstart and ystart are 1-based, and if they are 1 then it's a full frame. The shift of -1 puts them in zero-based ast reference that the rest of the pipeline is using..  ....Making sure I understand the logic flow correctly?

Reply:
Correct. And now if both are None or 1 in FITS, then a transform is not returned.
If only one is present a Shift is returned for this one. The other transform is `Identity` instead of `Shift(0)` (because it's a little faster).